### DISCUSSION TITLE
New parser question

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#5794] [General]
Hi

When running `'version' => 'regex',`, this set up works, but doesn't work with `'version' => 'runtime'`

```
{{ partial:_components/glide/placeholder/img glide_handle="url" pick_img="{{ index }}" img_ratio="16-9" }}
```

This is the `_components/glide/placeholder/img` partial
```
{{ placeholder }}
    {{ assets[pick_img] }}   
        {{ partial:_components/glide/ratio/{{img_ratio}} }}
    {{ /assets[pick_img] }}
{{ /placeholder }}
```

I get this error, the `img_ratio="16-9"` isn't working?
```
View [_components.glide.ratio.int_2caf9c2b0] not found.
```

Using `'version' => 'runtime'` it only works if I hardcode the value, but I'd like to still be able to pass that value in via a partial variable.
```
{{ placeholder }}
    {{ assets[pick_img] }}   
        {{ partial:_components/glide/ratio/16-9 }}
    {{ /assets[pick_img] }}
{{ /placeholder }}
```
I've tried different versions
```
{{ partial:_components/glide/ratio/{img_ratio} }}
```
```
{{ partial:_components/glide/ratio/'img_ratio' }}
```

I guess there must be a new syntax for this in 3.3, but I can't see any mention of this in the updated docs?

```
Statamic 3.3.5 Pro
Laravel 9.5.1
PHP 8.0.8
```

**Comment by duncanmcclean** [comment]
This sounds like a bug, the syntax shouldn't be any different between parsers. Can you open an issue instead?

**Comment by jackmcdade** [comment]
When you’re inside a pair of double braces (like a parameter), always use single braces to represent variables or tags to be parsed. 

Doubles sometimes work in regex but not always so we always have documented to use singles. Try this. 

`{{ partial:_components.glide.placeholder.img glide_handle="url" pick_img="{index}" img_ratio="16-9" }} `

**Comment by jasonvarga** [comment]
This bit is probably your issue. You can't use braces like this for the tag name itself. It was not a supported feature in the regex parser, and only worked by fluke.

```
{{ partial:_components/glide/ratio/{{img_ratio}} }}
```

You should use the src parameter instead, and use single braces inside it:

```
{{ partial src="_components/glide/ratio/{img_ratio}" }}
```

**Reply by stephenmeehanuk** [reply]
> `{{ partial src="_components/glide/ratio/{img_ratio}" }}`

This works. Thanks, I'll review any other partials that use double brackets. I've moved over to the new parser now.

**Reply by jasonvarga** [reply]
We're going to add support for the syntax you used though, since apparently a lot of people were using it.
